ic-os/components/setupos-scripts/install-hostos.sh

```diff
@@ -9,17 +9,23 @@ PATH="/sbin:/bin:/usr/sbin:/usr/bin"
 source /opt/ic/bin/functions.sh
 
 function install_hostos() {
-    echo "* Installing HostOS disk-image..."
-
     target_drive=$(find_first_drive)
+    echo "* Installing HostOS disk-image to ${target_drive}..."
 
     TMPDIR=$(mktemp -d)
+    # Release RAM.  Cannot be run concurrently with install-guestos.sh.
+    rm -f disk.img
+    # Extract the disk image to RAM.
+    echo "* Temporarily extracting the HostOS image to memory; please stand by for a few seconds"
     tar xafS /data/host-os.img.tar.zst -C "${TMPDIR}" disk.img
-
-    size=$(wc -c <"${TMPDIR}/disk.img")
-    size="${size:=0}"
-
-    pv -f -s "$size" "${TMPDIR}/disk.img" | dd of="/dev/${target_drive}" bs=10M conv=sparse
+    log_and_halt_installation_on_error "${?}" "Unable to extract HostOS disk-image."
+    # Duplicate the image to the disk.
+    # Progress is handled by status=progress.
+    # Makes a huge difference when running the setup under QEMU with no KVM.
+    # dd will detect nulls in chunks of 4M and sparsify the writes.
+    # In *non-KVM-accelerated* VM, this goes 500 MB/s, three times as fast as before.
+    echo "* Writing the HostOS image to /dev/${target_drive}"
+    dd if="${TMPDIR}/disk.img" of="/dev/${target_drive}" bs=4M conv=sparse status=progress
```

Comment:
I'm really curious: what is the block size based on? Above it's 10M, here 4M (and previously it was 10M).

Reply:
I chose a value that seemed to yield good performance, and maximize opportunities for sparsification by dd.